code/modules/antagonists/devil/imp/imp.dm

```diff
@@ -41,23 +43,13 @@
 							of intentionally harming a fellow devil.</B>"
 
 /mob/living/simple_animal/imp/Initialize()
-	..()
-	boost = world.time + 30
-
-/mob/living/simple_animal/imp/Life()
-	..()
-	if(boost<world.time)
-		speed = 1
-	else
-		speed = 0
+	. = ..()
+	set_varspeed(1)
+	addtimer(CALLBACK(src, /mob/living/simple_animal/proc/set_varspeed, 0), 30)
 
 /mob/living/simple_animal/imp/death()
-	..(1)
-	playsound(get_turf(src),'sound/magic/demon_dies.ogg', 200, 1)
-	visible_message("<span class='danger'>[src] screams in agony as it sublimates into a sulfurous smoke.</span>")
-	ghostize()
-	qdel(src)
-
+	playsound(src,'sound/magic/demon_dies.ogg', 80, 1)
```

Comment:
change the 4th parameter to TRUE it's a boolean

Reply:
Well I already did something else before I saw this.